File: 1700-minimum-time-to-make-rope-colorful/minimum-time-to-make-rope-colorful.cpp
class Solution {
public:
    int minCost(string colors, vector<int>& neededTime) {
        int t = 0;
        int m = neededTime[0];
        for(int i=1;i<colors.length();i++){
            if(colors[i] != colors[i-1]){
                m = neededTime[i];
            }
            else{
                t += min(m,neededTime[i]);
                m = max(m,neededTime[i]);
            }
        }
        return t;
    }
};

File: 1700-minimum-time-to-make-rope-colorful/minimum-time-to-make-rope-colorful.py
class Solution:
    def minCost(self, colors: str, neededTime: List[int]) -> int:
        t, c = 0, 0
        for i in range(len(colors)):
            if colors[i] != colors[i - 1]:
                c = 0
            t += min(c, neededTime[i])
            c = max(c, neededTime[i])
        return t